### DISCUSSION TITLE
AdvantEDGE Setup on VM

### DISCUSSION THREAD

**Discussion by dilallkx** [InterDigitalInc/AdvantEDGE#437] [Q&A]
From @sunilaradhya

_Hi Roy,
Could you please help me how to setup Advantedge on Ubuntu VM. Have some doubts. Is any hardware required for this setup._

**Comment by dilallkx** [comment]
Moving discussion to this new thread.

**Comment by dilallkx** [comment]
Hi @sunilaradhya,

Thank you for reaching out!
Please have a look at the setup pages in the [AdvantEDGE Docs](https://interdigitalinc.github.io/AdvantEDGE/docs/setup). Please let me know if you have any issues while installing the AdvantEDGE using these instructions.

Regards!

**Comment by dilallkx** [comment]
From @sunilaradhya

_Hi,
I setup the advatedge on one ubuntu system and now I am able to access the advatedge GUI using node ip.
If I want to deploy web application on this AdvantaEDGE GUI what are the steps.
Please help on this ._

_Regards_

**Comment by dilallkx** [comment]
Hi @sunilaradhya,

Demo1 example scenario is a good place to start to see how a web application can be deployed in AdvantEDGE. The _demo-svc_ application in Demo1 has a frontend component that can be accessed from a browser when deployed. I recommend taking a closer look at the Demo1 example:
- Demo1 [documentation](https://interdigitalinc.github.io/AdvantEDGE/docs/usage/usage-demo1/)
- Demo1 [src](https://github.com/InterDigitalInc/AdvantEDGE/tree/master/examples/demo1)

I hope this helps you get started.
Regards!

**Reply by sunilaradhya** [reply]
Hi Kevin,

Thanks for you reply. I setup AdvantEDGE on one VM and also attached worker node to the cluster.  Now I have one master and one worker node on K8S cluster. But AdvantEDGE setup on master and I am accessing AdvantEDGE GUI using master node ip on browser. 

Actually my requirement was I need to deploy web application on worker nodes which is joined the cluster. I need to configure and control those nodes (VMs) from Master/Control node(i.e. from UI). Is this Feature/option available on AdvantEDGE GUI. 
I need to orchestrate those worker nodes from master to deploy or add /remove application.  This Feature supports AdvantEDGE?
For Example AdvantEDGE GUI accessing on IP of  Master node of K8S cluster. If I want to configure/schedule and remove  my custom application on some worker nodes IP , How to do that. I didn't find any provision to configure nodes on AdvantEDGE GUI

Also have doubt on How to deploy web application on worker node from master using UI. 
If I want to deploy custom application using Advantedge how to do that.
Please help me on this.

Regards

**Reply by sunilaradhya** [reply]
Hi Kevin,

Can you please reply on my Previous comments

Regards

**Reply by dilallkx** [reply]
Hi @sunilaradhya,

Apologies for the delayed responses!

AdvantEDGE allows you to create a scenario where you can configure an emulated network for your applications. You can deploy your applications by creating container images and including them in the scenario to be deployed on the platform. Demo1 provides a good example of how to do so by including UE and Edge applications in the demo1 scenario.

Once your scenario is saved and deployed, your configured application containers should start executing on the platform. You should see the k8s pods come up with your applications. If you have services running in your application containers, you can access them directly from other scenario applications, or you can expose them to external applications using a k8s NodePort (configurable in scenario as explained [here](https://interdigitalinc.github.io/AdvantEDGE/docs/usage/gui/config-view/#external-port)).

WRT your question about dynamically orchestrating applications on the worker nodes, this is possible with AdvantEDGE. You must create an deploy a scenario where your application is not configured. You can then sent a _compute_ event to the platform to add a process. This feature is documented [here](https://interdigitalinc.github.io/AdvantEDGE/docs/overview/features/events/) and [here](https://interdigitalinc.github.io/AdvantEDGE/docs/overview/features/process-lifecycle/), and the UI controls are explained [here](https://interdigitalinc.github.io/AdvantEDGE/docs/usage/gui/exec-view/#scenario-update). To make sure this process runs on the worker node, you can use the pod placement feature described [here](https://interdigitalinc.github.io/AdvantEDGE/docs/overview/features/pod-placement/).

I hope this helps answer your questions and provides you with the information you need to deploy your application successfully in AdvantEDGE.

Best regards!

**Reply by sunilaradhya** [reply]
Hi Kevin, 
Thanks for the response.
I tried before this way which you provided the solution. Still I didnt get clarity. Please help me how can I create scenario to create an application like WordPress and nginx. Help me with two applications for scenario step by step. So it helps me better understand.

Also I deployed AdvantaEDGE successfully on one machine and I am getting green indicator on the AdvantaEDGE UI. After that Is I need to do any changes. I tried to deploy all the example scenario. I configured correctly and saved it correctly. In the execute window I Create a sandbox name. Then only Deploy option enable. when I click on deploy option it will open drop down menu, In that saved scenario name will pop up and click on ok. so, it will deployed. Then I am not able to see the pods are running in Advantedge installed VM. Also except Deploy option button other buttons are not enabled. Is I need to configure anything please help me on this also.

Regards
Sunil

**Reply by dilallkx** [reply]
Hi @sunilaradhya,

It seems you are getting an error during scenario deployment. Either your scenario configuration is  incorrect or one of your scenario pods must not be starting correctly. You can use the command `kubectl logs meep-virt-engine-...` to obtain logs from the Virtualization Engine to check for an error during scenario deployment. With these logs you will be able to see if and why a helm installation failed for one of your scenario pods.

Also, if your scenario pods are deployed but not starting properly you can use the command `kubectl describe pods -n <sandbox-name> <pod-name>` to get information about the pod deployment status.

I hope this helps pinpoint your issues!
Regards.
